source/plugin/data/serialization.rst

```diff
@@ -123,3 +125,51 @@ from persistent files using the :doc:`Configurate Library <../configuration/inde
         final DataView container = translator.translate(node);
         return deserializeHealth(container);
     }
+
+
+DataFormat
+==========
+An alternative to DataTranslators is to use :javadoc:DataFormat, which allows you to store a ``DataContainer`` in HOCON, JSON or NBT format. You can also recreate DataConainters using ``DataFormats``.
+
+For example, you could use ``DataFormat`` to create a json repusentation of a ``DataContainer``, which you could then easily store in a database as JSON. ``DataFormat`` can then be used to recreate the original ``DataContainer`` from json where required.
+
+For this example we will use :javadoc:`DataFormats#JSON` to translate a ItemStackSnapshot to JSON and back! 
+
+**Imports for code examples**
+    import org.spongepowered.api.Sponge;
+    import org.spongepowered.api.data.DataContainer;
+    import org.spongepowered.api.data.DataQuery;
+    import org.spongepowered.api.data.DataView;
+    import org.spongepowered.api.data.persistence.DataBuilder;
+    import org.spongepowered.api.data.persistence.DataFormat;
+    import org.spongepowered.api.data.persistence.DataFormats;
+    import org.spongepowered.api.item.inventory.ItemStackSnapshot;
+
+    import java.io.ByteArrayOutputStream;
+    import java.io.IOException;
+    import java.util.Optional;
+
+**Code Example: Serializing a ItemStackSnapshot to JSON format**
+    
+.. code-block:: java
+
+    public ByteArrayOutputStream toJsonFormat(ItemStackSnapshot itemStack){
+        DataContainer playerDataContainer = itemStack.toContainer();
+        DataFormat dataFormat = DataFormats.JSON;
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        try {
+            dataFormat.writeTo(outputStream, playerDataContainer);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return outputStream;
+    }
+
+**Code Example: Deserializing a ItemStackSnapshot from json format**
```

Comment:
So, the original thought of mine is that while it doesn't matter really what we deserialise to, having an example in context is often beneficial - an `ItemStackSnapshot` is a classic example of what you might want to serialise/deserialise. Right now, this code example is not useful, because it now skips the step of _exactly_ what we want to show.

Effectively, the goal is to say "hey, if you don't want to serialise your `DataContainer` into configuration, then `DataFormat` is for you". I hate to say this, but many developers learn by example, and may be stuck on the point that `ByteArrayOutputStream`s and input streams are here, I suspect some don't know what they do and is going to detract from the actual goal. I also expect most developers will use this for databases, and just want to get a string out to save into the database. 

On the flip side, saying "here is a data container, here is JSON, here is the container again" is not going to wash either - why would a developer care about a simple data container?

So, how about, rather than providing two examples, we consider providing a sample class here that developers can use that is essentially a template on how we might want to store and retrieve a data container backed object? Something like `JsonItemStackSnapshotConverter` (or whatever you like) that contains two commented methods, `toJson` and `fromJson`, that illustrates how to get a `String` from an `ItemStackSnapshot`, and vice versa. That way, we get the context of when we might use this (item serialisation), and how this would be performed?